src/dune_rules/lib_rules.ml

```diff
@@ -570,86 +571,112 @@ let library_rules
   let* requires_compile = Compilation_context.requires_compile cctx in
   let ocaml = Compilation_context.ocaml cctx in
   let stdlib_dir = ocaml.lib_config.stdlib_dir in
-  let top_sorted_modules =
-    let impl_only = Modules.impl_only modules in
-    Dep_graph.top_closed_implementations
-      (Compilation_context.dep_graphs cctx).impl
-      impl_only
-  in
-  let* () =
-    Memo.Option.iter vimpl ~f:(Virtual_rules.setup_copy_rules_for_impl ~sctx ~dir)
-  in
-  let* () = Check_rules.add_cycle_check sctx ~dir top_sorted_modules in
-  let* () = gen_wrapped_compat_modules lib cctx
-  and* () = Module_compilation.build_all cctx
-  and* expander = Super_context.expander sctx ~dir
-  and* lib_info =
+  let* lib_info =
```

Comment:
are there unpredictable consequences of getting the lib info before doing the cycle check on its modules?